libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -838,10 +835,8 @@ void AP_TECS::_update_throttle_with_airspeed(void)
 float AP_TECS::_get_i_gain(void)
 {
     float i_gain = _integGain;
-    if (_flight_stage == AP_FixedWing::FlightStage::TAKEOFF || _flight_stage == AP_FixedWing::FlightStage::ABORT_LANDING) {
-        if (!is_zero(_integGain_takeoff)) {
-            i_gain = _integGain_takeoff;
-        }
+    if (_in_takeoff_first_stage() || _flight_stage == AP_FixedWing::FlightStage::ABORT_LANDING) {
+        i_gain = _integGain_takeoff;
```

Comment:
In effect his change means we are defaulting to an integral gain of 0 during takeoff whereas before it was 0.3. This is inconsistent with the parameter description  "When set to 0 then TECS_INTEG_GAIN is used. " and is also a change to the default tuning.

Most takeoff will spend little time with  _in_takeoff_first_stage true and the speed weighting is 2 so I would not expect it to have much effect for most applications. However there may be some low powered setups with large takeoff heights where the integrator will help the pitch angle to  converge to the value needed to maintain airspeed.

Please update the parameter documentation and change TECS_TKOFF_IGAIN to match the TECS_INTEG_GAIN value of 0.3 unless there was a problem with that value in which case we need to find a suitable default.

Reply:
Parameter description updated.